### PR TITLE
UE-2513: Fix limitation 4 GBs file in windows

### PR DESCRIPTION
**Pull request by quangnt**
Finish this. Please review

### CODE REVIEW

**Comment by quangnt** [comment]
@zelonght Address comments and also create process common methods for calling in filter_clean and filter_smudge. Please review. So ignore your changes request.

File: git-fat [hunk 1]
```diff
@@ -416,24 +418,49 @@ class GitFat(object):
         '''
         self.setup()
         # Set stdin and stdout to binary mode
-        sys.stdin = ensure_binary_mode(sys.stdin)
+        # Process for large file size in Windows
```
Comment (quangnt) [inline git-fat]: Solved.

**Comment by quangnt** [comment]
@zelonght Finish address comments and the feature works now. Can you review?

**Comment by quangnt** [comment]
@PhuongHNguyen To verify this PR, please do as following guideline.
[HOWTOVERIFY.txt](https://github.com/PersonifyInc/git-fat/files/1107831/HOWTOVERIFY.txt)

File: git-fat [hunk 2]
```diff
@@ -673,6 +680,38 @@ class GitFat(object):
...
+            if filter.__func__.is_large_file_size:
+                return True
+        return False
+    
+    def has_git_filter_file_param_on_window(self):
+        if sys.platform == 'win32' and len(sys.argv) > 2:
+            return True
+        return False
+        
+    def process_large_file(self, filter, smudge_file_name = None):
```
Comment (zelonght) [inline git-fat]: @quangnt Implementation is complex and not simple as I suggested an previous comments. I think we need discuss this.

File: git-fat [hunk 3]
```diff
@@ -118,7 +118,7 @@ except ImportError:
             raise RuntimeError("S3Backend requires boto.")
 
 BLOCK_SIZE = 4096
-
+LARGE_FILE_SIZE = 4 * 1024 * 1024 * 1024
```
Comment (zelonght) [inline git-fat]: LARGE_FILE_SIZE = 4294967296 # 4 * 1024 * 1024 * 1024 = 4GB

Comment (quangnt) [inline git-fat]: Solved.

**Comment by quangnt** [comment]
@PhuongHNguyen This PR is ready for testing.

**Comment by zelonght** [comment]
@PhuongHNguyen can you verify this in old git-fat machine (backward compatiblity) and new git-fat machine (to make sure new filter) work well in both cases.

Also, please report if this work with small (200 - 500MB) and large file (>4GB). Thanks!

**Comment by PhuongHNguyen** [comment]
👍 I verified this PRs with 4 cases:

- There's "%f" on "\.git\config" and I added a sequences ~ 300MB success

- There's "%f" on "\.git\config" and I added a sequences ~ 8.5GB success

- There's not "%f" on "\.git\config" and I added a sequences ~ 300MB  success

- There's not "%f" on "\.git\config" and I added a sequences ~ 8.5GB fail, but from TQuang, this issue is happened before

**Comment by quangnt** [comment]
@PhuongHNguyen Looks like you already check commit to branch then switch between branches to make sure new sequences are successfully added. For case not "%f", we only want to make sure that it works normally for file < 4GBs (as same behavior as before fixing) not affect by our changing.

**Comment by zelonght** [comment]
@PhuongHNguyen how about switch branch on pull sequences, is it restoring the file right ?

**Comment by PhuongHNguyen** [comment]
@zelonght : I switched branch and it is restored the file right.

File: git-fat [hunk 4]
```diff
@@ -569,9 +579,15 @@ class GitFat(object):
         if gitconfig_get('filter.fat.clean') or gitconfig_get('filter.fat.smudge'):
             print('Git fat already configured, check configuration in .git/config')
         else:
-            gitconfig_set('filter.fat.clean', 'git-fat filter-clean')
-            gitconfig_set('filter.fat.smudge', 'git-fat filter-smudge')
+            filter_clean = 'git-fat filter-clean'
```
Comment (zelonght) [inline git-fat]: @quangnt we don't need to seperate logic for this, add %f always will be good too.

**Comment by zelonght** [comment]
@PhuongHNguyen thank Phuong!
Merging...

**Pull request merged**